packages/core/http/core-http-server-internal/src/http_server.ts

```diff
@@ -385,6 +389,44 @@ export class HttpServer {
     }
   }
 
+  private getDeprecatedRoutes(): RouterDeprecatedRouteDetails[] {
+    const deprecatedRoutes: RouterDeprecatedRouteDetails[] = [];
+
+    for (const router of this.registeredRouters) {
+      const allRouterRoutes = [
+        // exclude so we dont get double entries.
+        // we need to call the versioned getRoutes to grab the full version options details
+        router.getRoutes({ excludeVersionedRoutes: true }),
+        router.versioned.getRoutes(),
+      ].flat();
+
+      deprecatedRoutes.push(
+        ...allRouterRoutes
+          .flat()
+          .map((route) => {
+            if (route.isVersioned === true) {
+              return Object.entries(route.versionsOptions).map(([version, handlerOptions]) => {
+                return { route, version, deprecated: handlerOptions.options?.deprecated };
```

Comment:
We should be able to do this instead, and then we don't need `versionedOptions`

```suggestion
return route.handlers.map(({ options: { version, options } }) => {
                return { route, version, deprecated: options?.deprecated };
```